dotcom-rendering/src/server/prod-server.ts

```diff
@@ -21,13 +21,20 @@ import { recordBaselineCloudWatchMetrics } from './lib/aws/metrics-baseline';
 import { getContentFromURLMiddleware } from './lib/get-content-from-url';
 import { logger } from './lib/logging';
 
+const getPageIdFromBody = (body: unknown): string => {
+	const baseBody = body as BaseRequestBody;
+	return baseBody.pageId || 'no-page-id-found';
```

Comment:
I guess you could do something along the lines of `return 'pageId' in baseBody ? baseBody.pageId : 'no-page-id-found';` instead of creating a new type?

Reply:
It seems TS doesn't like me doing `'x' in body` when body is `unkown`

Reply:
Oh really? that's annoying :/

I guess you might need to do something along the lines of `(body instanceof Object && 'pageId' in body)` or perhaps `(body && 'pageId' in body)`!

Reply:
Perhaps:

```ts
isObject(body) && typeof body.pageId === 'string' ? body.pageId : 'no-page-id-found';
```

where `isObject` comes from `@guardian/libs`.

There's also a `parser` module in AR that allows you to parse values from `unknown` objects like this:

```ts
const pageIdParser = fieldParser('pageId', stringParser);
```

But that's possibly overkill for something this small.

Reply:
Given the `body` we inherit from `Request` is of type `any`, could we write a guard?

```ts
const request: Record<string, any> = {}
const { body } = request; // body: any

function hasPageId(val: any): val is { pageId: string } {
  try {
    return 'pageId' in val;
  } catch(e) {
    return false;
  }
}

const pageId = hasPageId(body) ? body.pageId : 'no-page-id';
```